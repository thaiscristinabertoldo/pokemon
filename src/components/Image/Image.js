import React, { memo } from 'react'

import notfound from '../../assets/notfound.png'

export const Image = memo(({ src, name = 'Image', size = '250px', ...props }) => {
  return (
    <img src={src ? src : notfound} height={size} width={size} title={name} alt={name} data-testid="image" {...props} />
  )
})
