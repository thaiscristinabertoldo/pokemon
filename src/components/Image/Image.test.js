import React from 'react'

import { render, cleanup } from '@testing-library/react'

import { Image } from './Image'

afterEach(cleanup)

describe('Image', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Image />)

    expect(asFragment(<Image />)).toMatchSnapshot()
  })

  it('should render a src', () => {
    const { getByTestId } = render(<Image src="https://img.pokemondb.net/artwork/bulbasaur.jpg" />)
    expect(getByTestId('image')).toHaveAttribute('src', 'https://img.pokemondb.net/artwork/bulbasaur.jpg')
  })
})
