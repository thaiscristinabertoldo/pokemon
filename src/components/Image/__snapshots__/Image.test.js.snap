// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should take a snapshot 1`] = `
<DocumentFragment>
  <img
    alt="Image"
    data-testid="image"
    height="250px"
    src="notfound.png"
    title="Image"
    width="250px"
  />
</DocumentFragment>
`;
